data_sync.R

```diff
@@ -3,17 +3,92 @@
 # of the data provided to us by Care Evolution(CE) in the input bucket.
 ##############
 ## Required functions and parameters
-source('awscli_utils.R')
-source('params.R')
+source('~/recover-s3-synindex/awscli_utils.R')
+source('~/recover-s3-synindex/params.R')
+library('synapser')
 
 #############
-# Sync S3 buckets 
+# Sync S3 ingress bucket to Local EC2 (First Sync )
+# using prod-creds (with 'read' only permissions) for accessing the ingress bucket 
 #############
-# <<values below are test values>>
-s3SyncBuckets(source_bucket = paste0('s3://', INGRESS_BUCKET,'/'),
-              destination_bucket = paste0('s3://', PRE_ETL_BUCKET,'/'))
+# synapser::synLogin(daemon_acc, daemon_acc_password) # login into Synapse
+synapser::synLogin()
+
+sts_token <- synapser::synGetStsStorageToken(entity = 'syn52293299', # sts enabled destination folder
+                                             permission = 'read_write',  
+                                             output_format = 'json')
+
+# configure the environment with AWS token (this is the aws_profile named 'env-var')
+Sys.setenv('AWS_ACCESS_KEY_ID'=sts_token$accessKeyId,
+           'AWS_SECRET_ACCESS_KEY'=sts_token$secretAccessKey,
+           'AWS_SESSION_TOKEN'=sts_token$sessionToken)
+
+# access AWS s3 from env set sts token, sts token was requested from Synapse folder above
+s3SyncToLocal(source_bucket = paste0('s3://', INGRESS_BUCKET,'/'),
+              local_destination = AWS_DOWNLOAD_LOCATION,
+              aws_profile = 'env-var') 
+
 #############
-# Sync the pre-ETL bucket to local EC2 instance
+# Rename folders with '\' to have '_' for eg., 'adults\v1' to 'adults_v1'
 #############
-# <<values below are test values>>
-s3SyncToLocal(source_bucket = paste0('s3://', PRE_ETL_BUCKET,'/'), local_destination = AWS_DOWNLOAD_LOCATION)
+# We will work with the locally replicated structure 
+# all folders inside the AWS_DOWNLOAD_LOCATION, i.e all folders at AWS_DOWNLOAD_LOCATION/
+localDirs <- list.dirs(path = AWS_DOWNLOAD_LOCATION, full.names = FALSE, recursive = FALSE) 
+
+# The folders are named as 'adults\v1', 'pregnant\v1' and 'pediatric\v1'. We want to remove the
+# '\' and replace it with '_'
+for(dir_ in localDirs){
+  dir_newName <- stringr::str_replace_all(dir_, "\\\\",'_')
+  # rename each folder
+  file.rename(paste0(AWS_DOWNLOAD_LOCATION,"/",dir_), paste0(AWS_DOWNLOAD_LOCATION,"/",dir_newName))
+}
+
+# Get newly renamed folders
+localDirs <- list.dirs(path = AWS_DOWNLOAD_LOCATION, full.names = FALSE, recursive = FALSE) 
+
+
+#############
+# Sync Local EC2 (from ingress bucket) to pre_etl bucket (Second Sync )
+#############
+# synapser::synLogin(daemon_acc, daemon_acc_password) # login into Synapse
+synapser::synLogin()
+
+sts_token <- synapser::synGetStsStorageToken(entity = 'syn51714264', # sts enabled destination folder
```

Comment:
syn51714264 - Is the folder for ce-input-data attached to s3://recover-input-data/main

LOOKS GOOD